File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/Java_Gradle_Project/Laboratory_1/FestivalTicketSale/src/main/java/org/example/Model/Validators/ModelValidators/AngajatValidator.java
package org.example.Model.Validators.ModelValidators;

import org.example.Model.Exceptions.ValidatorException;
import org.example.Model.Angajat;
import org.example.Model.Validators.Interfaces.Validator;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class AngajatValidator implements Validator<Angajat> {
    private static final String EMAIL_REGEX =
            "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";

    private static final Pattern pattern = Pattern.compile(EMAIL_REGEX);

    public static boolean isValidEmail(String email) {
        Matcher matcher = pattern.matcher(email);
        return matcher.matches();
    }
    @Override
    public void validate(Angajat entity) throws ValidatorException {

        String errors="";
        if(entity.getNume().isEmpty() || entity.getNume().length()<=2)
        {
            errors+="Name must be at least 2 characters!\n";
        }
        if(entity.getPrenume().isEmpty() || entity.getPrenume().length()<=2)
        {
            errors+="Prenumele must be at least 2 characters!";
        }
        if(!isValidEmail(entity.getEmail()))
        {
            errors+="Email is not valid!";
        }
        if(entity.getUsername().length()<3)
        {
            errors+="Username is not valid! It must be at least 3 characters!";
        }
        if(entity.getPassword().length()<3)
        {
            errors+="Password must be at least 2 characters long!";
        }
        if(!errors.isEmpty())
        {
            throw new ValidatorException(errors);
        }

    }
}

File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/C#_Project/Laboratory_8_Proto/FestivalTicketSaleCS/model/Validators/ModelValidators/AngajatValidator.cs
using System;
using System.Text.RegularExpressions;
using model.Exceptions;
using model.Validators.Interfaces;


namespace model.Validators.ModelValidators
{
    public class AngajatValidator : IValidator<Angajat>
    {
        private static readonly string EMAIL_REGEX =
            @"^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,7}$";

        private static readonly Regex pattern = new Regex(EMAIL_REGEX);

        public static bool IsValidEmail(string email)
        {
            return pattern.IsMatch(email);
        }

        public void Validate(Angajat entity)
        {
            string errors = "";

            if (string.IsNullOrEmpty(entity.Nume) || entity.Nume.Length <= 2)
            {
                errors += "Name must be at least 2 characters!\n";
            }
            if (string.IsNullOrEmpty(entity.Prenume) || entity.Prenume.Length <= 2)
            {
                errors += "Prenumele must be at least 2 characters!";
            }
            if (!IsValidEmail(entity.Email))
            {
                errors += "Email is not valid!";
            }
            if (entity.Username.Length < 3)
            {
                errors += "Username is not valid! It must be at least 3 characters!";
            }
            if (entity.Password.Length < 3)
            {
                errors += "Password must be at least 2 characters long!";
            }
            if (!string.IsNullOrEmpty(errors))
            {
                throw new ValidatorException(errors);
            }
        }

        public void validate(Angajat entity)
        {
            throw new NotImplementedException();
        }
    }
}